runtime/qsym_backend/Runtime.cpp

```diff
@@ -321,6 +340,7 @@ void _sym_push_path_constraint(SymExpr constraint, int taken,
   if (constraint == nullptr)
     return;
 
+  Tracer::tracePathConstraint(constraint, taken != 0);
```

Comment:
Shouldn't be calls to tracer put into compiler plugin code? So that it will be runtime independent?

Reply:
I don't get it. What do you mean by compiler plugin code ?

Reply:
`compiler` directory in symcc repository root (https://github.com/eurecom-s3/symcc/tree/master/compiler). There is a code, which is executed on compilation phase as a plugin to clang.

Reply:
Imho it should also not be put in `compiler`, otherwise, SymQEMU will not be able to use tracing.
I guess the best option would be to have every function listed in `RuntimeCommon.h` wrapped into a runtime-agnostic function calling to the tracer and the runtime-specific corresponding operation.
The extra call indirection should be optimized out at compile-time.